### PR TITLE
Add syntax support for Scala & Twitter Futures in Mapper

### PR DESCRIPTION
**Pull request by sergeykolbasov**
Related to #1013

### CODE REVIEW

**Comment by vkostyukov** [comment]
I like it @sergeykolbasov! What's your thoughts on how long we have to keep these around? For the record, I don't think  "indefinitely" is unreasonable.

File: core/src/main/scala/io/finch/internal/package.scala [hunk 1]
```diff
@@ -121,4 +122,9 @@ package object internal {
       new String(array, begin, end - begin, cs.name)
     }
   }
+
+  private[internal] val dummieExecutionContext: ExecutionContext = new ExecutionContext {
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/internal/package.scala]: I'd probably prefer this being its own file under `internal` package (perhaps as an `object`). Also no need to limit its visibility as it has `internal` in its name.

**Comment by sergeykolbasov** [comment]
I think as long as we support the previous endpoint version.
I'm going to add `deprecated` flag to those implicit conversions to throw some warnings during compile phase

File: core/src/main/scala/io/finch/internal/Mapper.scala [hunk 2]
```diff
@@ -106,11 +181,37 @@ object Mapper extends HighPriorityMapperConversions {
...
+  implicit def mapperFromFutureResponseHFunction[F[_] : Effect, A, FN, FR](f: FN)(implicit
+    ftp: FnToProduct.Aux[FN, A => FR],
+    ev: FR <:< TwitterFuture[Response]
+  ): Mapper.Aux[F, A, Response] = instance(_.mapOutputAsync { value =>
+    val fr = twitterFutureToEffect(ev(ftp(f)(value)))
+    fr.map(r => Output.payload(r, r.status))
+  })
+
+  implicit def mapperFromScFutureOutputHFunction[F[_] : Effect, A, B, FN, FOB](f: FN)(implicit
+                                                                                    ftp: FnToProduct.Aux[FN, A => FOB],
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/internal/Mapper.scala]: Formatting here is somewhat crazy.

File: core/src/main/scala/io/finch/internal/Mapper.scala [hunk 3]
```diff
@@ -106,11 +181,37 @@ object Mapper extends HighPriorityMapperConversions {
...
+                                                                                    ev: FOB <:< ScalaFuture[Output[B]]
+  ): Mapper.Aux[F, A, B] = instance(_.mapOutputAsync(value => scalaFutureToEffect(ev(ftp(f)(value)))))
+
+  implicit def mapperFromScFutureResponseHFunction[F[_] : Effect, A, FN, FR](f: FN)(implicit
+                                                                                  ftp: FnToProduct.Aux[FN, A => FR],
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/internal/Mapper.scala]: Here as well.

Comment (sergeykolbasov) [reply core/src/main/scala/io/finch/internal/Mapper.scala]: I hope we gonna have scalafmt soon 😞

Comment (vkostyukov) [reply core/src/main/scala/io/finch/internal/Mapper.scala]: Yea that would be rad. I've seen cats recently adopted it too. Sounds like we're going to be a good company after all.

**Pull request merged**